examples/rifle_sxiv.sh

```diff
@@ -11,38 +11,38 @@
 #   mime ^image, has sxiv, X, flag f = path/to/this/script -- "$@"
 #
 # Implementation notes: this script is quite slow because of POSIX limitations
-# and portability concerns. First calling the shell function 'abspath' is
+# and portability concerns. First, using case statement to get absolute path is
 # quicker than calling 'realpath' because it would fork a whole process, which
 # is slow. Second, we need to append a file list to sxiv, which can only be done
 # properly in two ways: arrays (which are not POSIX) or \0 sperated
 # strings. Unfortunately, assigning \0 to a variable is not POSIX either (will
 # not work in dash and others), so we cannot store the result of listfiles to a
 # variable.
 
-if [ $# -eq 0 ]; then
-    echo "Usage: ${0##*/} PICTURES"
-    exit
-fi
-
-[ "$1" = '--' ] && shift
+listfiles () {
+    find -L "///${target%/*}" -maxdepth 1 -type f -iregex \
+      '.*\.\(jpe?g\|png\|gif\|webp\|tiff\|bmp\)$' -print0 | sort -z
```

Comment:
> My find(1) on a mac defaults to BREs which don't support `?` and `|`. Maybe we should just enable EREs?

I'm on gnu find and `-LE` gives an error.
```sh
$ find: unknown predicate `-LE'
```
Also how much do we care about posix compatibility? Because `-print0` isn't posix. OpenBSD man page does list `-print0` as an option however.

Also sxiv is strictly X, can X11 software be run on a mac? If not then we shouldn't care about it.

> maybe we should add `fd`

I don't personally have it (or ripgrep) on my system but I'm fine adding support for it. Although I would prefer using variables for the command instead of `||`.
Also if it's faster, shouldn't we try to use it as default and fallback to find instead?

Reply:
On the topic on posix, I don't think `xargs -0` is posix either.

Also... if we go with the tee option described in https://github.com/ranger/ranger/pull/2411#issuecomment-917865113 , we may not need null separator at all. Sxiv has an `-i` flag to read filenames from stdin. 

```diff
-    xargs -0 sxiv $args -n "${count%%:*}" < "$tmp" ||
+    sxiv $args -n "${count%%:*}" -i < "$tmp" ||
```
It seems to work fine on names with spaces as well.